src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -47,20 +47,24 @@ export class RawSqlResultsToEntityTransformer {
      * Groups given raw results by ids of given alias.
      */
     protected group(rawResults: any[], alias: Alias): any[][] {
-        const groupedResults: { id: any, items: any[] }[] = [];
+        const groupedResults: { [key: string]: { id: any, items: any[] } } = {};
         rawResults.forEach(rawResult => {
             const id = alias.metadata.primaryColumns.map(column => rawResult[alias.name + "_" + column.databaseName]).join("_"); // todo: check partial
             if (!id) return;
 
-            let group = groupedResults.find(groupedResult => groupedResult.id === id);
-            if (!group) {
+            let group: { id: any, items: any[] };
+            if (id in groupedResults) {
+                group = groupedResults[id];
+            } else {
                 group = { id: id, items: [] };
-                groupedResults.push(group);
             }
 
             group.items.push(rawResult);
         });
-        return groupedResults.map(group => group.items);
+
+        return Object.keys(groupedResults).map(function(key, index) {
```

Comment:
Also when doing persistence optimizations I found out `Object.keys` is expensive operation. If you are able to test it with significantly more data (hundred of millions) I think you'll found Object.keys price as well. Alternative to this method is store keys in the array in a method above.

Reply:
also you can use regular `=>` function and you don't need an `index` here.